include/votca/csg/topology.h

```diff
@@ -441,9 +441,9 @@ inline Bead *Topology::CreateBead(Bead::Symmetry symmetry, std::string name,
                                   std::string type, Index resnr, double m,
                                   double q) {
 
-  Bead *b = new Bead(_beads.size(), type, symmetry, name, resnr, m, q);
-  _beads.push_back(b);
-  return b;
+  _beads.emplace_back(
+      new Bead(_beads.size(), type, symmetry, name, resnr, m, q));
```

Comment:
You make a copy here now. Before they just copied the pointer, I think?

Reply:
I'm confused are we looking at the same lines of code :). They were calling `new Bead` so ya they were storing the pointer in the vector but the memory was on the heap and that vector is responsible for the memory management of the bead.

Reply:
I did not look at the line above. YOu are right.